src/dune_rules/lib_id.ml

```diff
@@ -18,7 +23,7 @@ module Local = struct
       | x -> x
     ;;
```

Comment:
I missed this from before, but this seems wrong. How is the comparison function ignoring the other elements?

Reply:
My thought was that, since this refers to `Local` library IDs, the `loc` would be more than enough to distinguish different lib_ids.

Reply:
The only field we're not comparing is `enabled_if`. We could, but I couldn't come up with a `compare` function for `Blang.t` -- there's only `Blang.equal`.

Reply:
> My thought was that, since this refers to Local library IDs, the loc would be more than enough to distinguish different lib_ids.

In that case, I would drop the `enabled_if` from the id as well.